Formula/xinit.rb

```diff
@@ -0,0 +1,117 @@
+class Xinit < Formula
+  desc "Start the X Window System server"
+  homepage "https://gitlab.freedesktop.org/xorg/app/xinit"
+  url "https://www.x.org/releases/individual/app/xinit-1.4.1.tar.bz2"
+  sha256 "de9b8f617b68a70f6caf87da01fcf0ebd2b75690cdcba9c921d0ef54fa54abb9"
+  license all_of: ["MIT", "APSL-2.0"]
+
+  depends_on "pkg-config" => :build
+  depends_on "tradcpp"    => :build
+
+  depends_on "libx11"
+  depends_on "lndir"
+  depends_on "mkfontscale"
+  depends_on "quartz-wm"
+  depends_on "xauth"
+  depends_on "xmodmap"
+  depends_on "xorg-server"
+  depends_on "xrdb"
+  depends_on "xterm"
+
+  resource "font_cache" do
+    url "https://raw.githubusercontent.com/XQuartz/XQuartz/XQuartz-2.8.1/base/opt/X11/bin/font_cache"
+    sha256 "8f27cf55e5053686350fce6ea1078e314bd5b2d752e9da1b9051541e643e79d6"
+  end
+
+  resource "98-user.sh" do
+    url "https://raw.githubusercontent.com/XQuartz/XQuartz/XQuartz-2.8.1/base/opt/X11/etc/X11/xinit/xinitrc.d/98-user.sh"
+    sha256 "b417aea949931b7c03133535c3b5146b9403b8c3482a1c1d0a5dc01c07876c84"
+  end
+
+  def install
+    configure_args = std_configure_args + %W[
+      --with-bundle-id-prefix=homebrew.mxcl
+      --with-launchagents-dir=#{prefix}
+      --with-launchdaemons-dir=#{prefix}
+      --with-xserver=#{Formula["xorg-server"].opt_bin}/X
+    ]
+
+    system "./configure", *configure_args
+    system "make", "RAWCPP=tradcpp"
+    system "make", "install"
+
+    inreplace bin/"startx", prefix, HOMEBREW_PREFIX
+    inreplace bin/"startx", HOMEBREW_PREFIX/"libexec", opt_libexec
+    inreplace prefix/"etc/X11/xinit/xinitrc", prefix, opt_prefix
```

Comment:
Sounds like stuff like `--with-xinitdir=` would cover this? Not entirely sure I entirely follow the need for opt prefixes here though.

Reply:
Deleted because it doesn't affect using `xinit`.

Reply:
Well, I'm referring to the `startx` inreplace too.

Reply:
If we drop `inreplace` in `startx`, `X11.app` will restart infinitely when users open it in finder, because it cannot find `xauth` in when executing `startx`. There is circular dependency between `xinit` and `xserver`.

`PATH` in `X11.app`:
```
PATH=/usr/local/Cellar/xinit/1.4.1/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

Reply:
So what lines exactly are you trying to replace in `startx`? It's not particularly clear from this.

Reply:
Changed line:

```diff
--- ./startx	2022-06-05 08:37:11.000000000 +0800
+++ /usr/local/bin/startx	2022-05-29 17:46:49.000000000 +0800
@@ -17,11 +17,11 @@
 # Note that we put our own bin directory at the front of the path, and
 # the standard system path at the back, since if you are using the Xorg
 # server there's a pretty good chance you want to bias the Xorg clients
 # over the old system's clients.
 # First our compiled path
-bindir=/usr/local/Cellar/xinit/1.4.1/bin
+bindir=/usr/local/bin
 case $PATH in
     *:$bindir | *:$bindir:* | $bindir:*) ;;
     *) PATH=$bindir:$PATH ;;
 esac 
```
I will add comment here.